db/index.js

```diff
@@ -22,41 +26,47 @@ const useJson = (() => {
 var DB = (() => {
 
   const validateCode = async (code) => {
-    code = code.toUpperCase()
 
     if (useJson) {
-      return await jsonDB.find(user => user.code === code) || null
+      return await jsonDB.find(user => verifyHash(code.toUpperCase(), user.code, {useInitialSalt: true})) || null
     }
 
+    code = hashString(code, {useInitialSalt: true}).catch(e => console.warn(e))
```

Comment:
Does this need to be upper-cased as well?

Reply:
no, because when we use the local db, it writes locally the hashed value into the session, and not the plaintext. So if we make it all caps the verify won't work